### PR TITLE
Tests implementation for SYCL2020 `host_accessor` constructors 

### PR DESCRIPTION
**Pull request by andreyromanof**
This provides tests for `host_accessor` constructors

Depends on:
- #318 
- #333 

[Link](https://github.com/KhronosGroup/SYCL-CTS/pull/325/files/6637d4d264accb65dc30fa6ff053705c6d3d8f5e..HEAD) to related changes

### CODE REVIEW

File: tests/accessor/host_accessor_constructors.h [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+  void operator()(const std::string& type_name) {
+    // Type packs instances have to be const, otherwise for_all_combination will
+    // not compile
+    const auto access_modes = get_access_modes();
+    const auto dimensions = get_dimensions();
+    const auto cur_type =
+        named_type_pack<T>::generate(type_name_string<T>::get(type_name));
+
+    for_all_combinations<run_tests_constructors>(cur_type, access_modes,
+                                                 dimensions);
```
Comment (yuriykoch) [inline tests/accessor/host_accessor_constructors.h:168]:
```suggestion
    for_all_combinations<run_tests_constructors, T>(access_modes, dimensions, type_name);
```
Will it work in a such way or we have a bottleneck here?

Comment (andreyromanof) [reply tests/accessor/host_accessor_constructors.h:168]: Looks like we need to update `for_all_combinations` in order to do this trick.

Comment (andreyromanof) [reply tests/accessor/host_accessor_constructors.h:168]: Upd: I was wrong. This actually works.

**Comment by keryell** [comment]
Waiting for https://github.com/KhronosGroup/SYCL-CTS/pull/318 to land to simplify the diff here.

**Comment by keryell** [comment]
There are some merge conflicts to resolve first.

File: tests/accessor/accessor_common.h [hunk 2]
```diff
@@ -358,14 +438,16 @@ void read_write_acc(AccT testing_acc, ResultAccT res_acc) {
  * @param r Range for accessors buffer
  */
 template <accessor_type AccType, typename DataT, int Dimension,
-          sycl::access_mode AccessMode, sycl::target Target,
+          sycl::access_mode AccessMode,
+          sycl::target Target = sycl::target::device,
```
Comment (yuriykoch) [inline tests/accessor/accessor_common.h:442]: Looks like we need the similar change for `check_zero_dim_constructor`

Comment (andreyromanof) [reply tests/accessor/accessor_common.h]: Fixed

File: tests/accessor/accessor_common.h [hunk 3]
```diff
@@ -283,14 +353,15 @@ void read_write_zero_dim_acc(AccT testing_acc, ResultAccT res_acc) {
  * @tparam GetAccFunctorT Type of functor for accessor creation
  */
 template <accessor_type AccType, typename DataT, sycl::access_mode AccessMode,
-          sycl::target Target, typename GetAccFunctorT>
+          sycl::target Target = sycl::target::device, typename GetAccFunctorT>
```
Comment (yuriykoch) [inline tests/accessor/accessor_common.h:356]: Nit: clang-format
```suggestion
          sycl::target Target = sycl::target::device,
          typename GetAccFunctorT>
```

Comment (andreyromanof) [reply tests/accessor/accessor_common.h:356]: Are you sure, that we need wrapping to a new line here? The length of line 356 is 79 and it is not triggering `clang-format`

Comment (yuriykoch) [reply tests/accessor/accessor_common.h:356]: Of course we do not, sorry)

I've had something like this locally due to a different base: https://github.com/KhronosGroup/SYCL-CTS/blob/1eeb805e6edaef53320121a9d1a54e1383757ace/tests/accessor/accessor_common.h#L462 
and somehow misinterpreted the direction of the local patch.

File: tests/accessor/host_accessor_constructors.h [hunk 4]
```diff
@@ -0,0 +1,179 @@
...
+constexpr accessor_type AccType = accessor_type::host_accessor;
+
+template <typename DataT, int Dimension, sycl::access_mode AccessMode>
+void test_default_constructor(const std::string& access_mode_name,
+                              const std::string& type_name) {
+  const auto section_name = get_section_name<Dimension>(
+      type_name, access_mode_name, "Default constructor");
+
+  SECTION(section_name) {
+    auto get_acc_functor = []() {
```
Comment (keryell) [inline tests/accessor/host_accessor_constructors.h:27]:
```suggestion
    auto get_acc_functor = [] {
```

**Review by keryell — APPROVED** [review]

File: tests/accessor/host_accessor_constructors.h (hunk 4 above)
Comment (andreyromanof) [reply tests/accessor/host_accessor_constructors.h]: Done

**Pull request merged**